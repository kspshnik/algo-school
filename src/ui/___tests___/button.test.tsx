/* eslint-disable react/react-in-jsx-scope */
import { screen, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from '../index';

// Константы для каждого теста в обоих сьютах
const buttonText = 'Test text!';
const buttonTestId = 'ui-regular-button';
const imgTestId = 'ui-button-loader-img';

describe('button should render according to props & fire onClick on click', () => {
  const imgAlt = 'Загрузка.';
  it('should render normal button w/text', () => {
    render(<Button text={buttonText} />);
    const button = screen.getByTestId(buttonTestId);
    expect(button).toHaveTextContent(buttonText);
  });
  it('should render normal button w/out text', () => {
    render(<Button />);
    const button = screen.getByTestId(buttonTestId);
    expect(button).toHaveTextContent('');
  });
  it('should render disabled', () => {
    render(<Button text='Test text!' disabled />);
    const button = screen.getByTestId(buttonTestId);
    expect(button).toBeDisabled();
  });
  it('should render loader', () => {
    render(<Button text='Test text!' isLoader />);
    const img = screen.getByTestId(imgTestId);
    expect(img).toHaveAttribute('alt', imgAlt);
  });
  it('should fire onClick on click', () => {
    const mockOnClick = jest.fn();
    render(<Button text='Test text!' onClick={mockOnClick} />);
    const button = screen.getByTestId(buttonTestId);
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});

describe('it should match snapshot', () => {
  it('should match normal button w/text', () => {
    const buttonSnapshot = renderer.create(<Button text={buttonText} />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it('should match normal button w/out text', () => {
    const buttonSnapshot = renderer.create(<Button />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it('should match disabled button', () => {
    const buttonSnapshot = renderer.create(<Button text={buttonText} disabled />).toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
  it('should match loader', () => {
    const buttonSnapshot = renderer.create(<Button text='Test text!' isLoader />);
    expect(buttonSnapshot).toMatchSnapshot();
  });
});
