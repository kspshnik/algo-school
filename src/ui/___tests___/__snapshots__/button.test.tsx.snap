// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot should match disabled button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="ui-regular-button"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    Test text!
  </p>
</button>
`;

exports[`it should match snapshot should match loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="ui-regular-button"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    data-testid="ui-button-loader-img"
    src="loader.svg"
  />
</button>
`;

exports[`it should match snapshot should match normal button w/out text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="ui-regular-button"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`it should match snapshot should match normal button w/text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="ui-regular-button"
  type="button"
>
  <p
    className="text undefined"
  >
    Test text!
  </p>
</button>
`;
